python/dgl/distributed/rpc_server.py

```diff
@@ -83,11 +88,20 @@ def start_server(server_id, ip_config, num_servers, num_clients, server_state, \
             # a new client group is ready
             ips.sort()
             client_namebook = dict(enumerate(ips))
+            time.sleep(3) # wait for clients' receivers ready
+            max_try_times = int(os.environ.get('DGL_DIST_MAX_TRY_TIMES', 120))
             for client_id, addr in client_namebook.items():
                 client_ip, client_port = addr.split(':')
-                # TODO[Rhett]: server should not be blocked endlessly.
+                try_times = 0
                 while not rpc.connect_receiver(client_ip, client_port, client_id, group_id):
+                    try_times += 1
+                    if try_times >= max_try_times:
+                        raise DGLError("Failed to connect receiver[{}:{}] after retry {} times. "
+                                       "Please check availability of target receiver or change max "
+                                       "try times via env 'DGL_DIST_MAX_TRY_TIMES'.".format(
```

Comment:
Failed to connect to receiver [{}:{}] after {} retries. Please check availability of this target receiver or change the max retry times via 'DGL_DIST_MAX_TRY_TIMES'.